ui/src/test/java/com/hyperwallet/android/ui/repository/UserRepositoryImplTest.java

```diff
@@ -0,0 +1,204 @@
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.ADDRESS_LINE_1), is("575 Market Street"));
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.CITY), is("San Francisco"));
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.STATE_PROVINCE), is("CA"));
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.COUNTRY), is("US"));
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.POSTAL_CODE), is("94105"));
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.LANGUAGE), is("en"));
+        assertThat(resultUser.getField(HyperwalletUser.UserFields.PROGRAM_TOKEN),
+                is("prg-83836cdf-2ce2-4696-8bc5-f1b86077238c"));
+    }
+
+    @Test
+    public void testLoadUser_returnEmptyUserData() {
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                HyperwalletListener listener = (HyperwalletListener) invocation.getArguments()[0];
+                listener.onSuccess(null);
+                return listener;
+            }
+        }).when(mHyperwallet).getUser(ArgumentMatchers.<HyperwalletListener<HyperwalletUser>>any());
+        UserRepository.LoadUserCallback mockCallback = mock(UserRepository.LoadUserCallback.class);
+
+        mUserRepository.loadUser(mockCallback);
+
+        verify(mockCallback).onUserLoaded(mUserCaptor.capture());
+        verify(mockCallback, never()).onError(any(HyperwalletErrors.class));
+
+        HyperwalletUser user = mUserCaptor.getValue();
+        assertThat(user, is(nullValue()));
+    }
+
+
+    @Test
+    public void testLoadUser_returnErrors() {
+
+        final HyperwalletError error = new HyperwalletError("test message", "TEST_CODE");
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                HyperwalletListener listener = (HyperwalletListener) invocation.getArguments()[0];
+                List<HyperwalletError> errorList = new ArrayList<>();
+                errorList.add(error);
+                HyperwalletErrors errors = new HyperwalletErrors(errorList);
+                listener.onFailure(new HyperwalletException(errors));
+                return listener;
+            }
+        }).when(mHyperwallet).getUser(ArgumentMatchers.<HyperwalletListener<HyperwalletUser>>any());
+        UserRepository.LoadUserCallback mockCallback = mock(
+                UserRepository.LoadUserCallback.class);
+
+        mUserRepository.loadUser(mockCallback);
+
+        verify(mockCallback, never()).onUserLoaded(ArgumentMatchers.<HyperwalletUser>any());
+        verify(mockCallback).onError(mErrorCaptor.capture());
+
+        assertThat(mErrorCaptor.getValue().getErrors(), hasItem(error));
+    }
+
+    @Test
+    public void testRefreshUser_callsGetUserAfterRefresh() {
```

Comment:
this test can be removed

Reply:
@fmattos-hw Done.

Reply:
@fmattos-hw I added it again. I have to add it for jacoco (without this test - only 50% was covered)